### PR TITLE
Pull deployment scripting and RBAC rule fixes into v0.5.0 release branch

### PR DESCRIPTION
**Pull request by davidz627**
/assign @msau42 @verult

### CODE REVIEW

**Comment by verult** [comment]
/lgtm
Assuming the other two commits are intentional since they were merged to master recently

File: deploy/kubernetes/base/setup-cluster.yaml [hunk 1]
```diff
@@ -54,12 +54,21 @@ rules:
   - apiGroups: ["storage.k8s.io"]
     resources: ["storageclasses"]
     verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
```
Comment (msau42) [inline deploy/kubernetes/base/setup-cluster.yaml]: do we want snapshot rules here or only in alpha overlay?

**Comment by verult** [comment]
This one is for master: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/284

**Comment by davidz627** [comment]
moved snapshot rules to alpha overlay

**Pull request merged**